### PR TITLE
Fix background initialization

### PR DESCRIPTION
**Pull request by pchien**
We were accessing UIApplication from a background thread.

https://github.com/mixpanel/mixpanel-iphone/issues/710

### CODE REVIEW

**Comment by irace** [comment]
Hi – can you please merge this? 🙏

File: HelloMixpanel/HelloMixpanelTests/MiscEdgeCaseTests.m [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#import <XCTest/XCTest.h>
+#import "MixpanelBaseTests.h"
+
+@interface MiscEdgeCaseTests : MixpanelBaseTests
+
+@end
+
+@implementation MiscEdgeCaseTests
+
+- (void)testInitializeMixpanelOnBackgroundThread {
```
Comment (yarneo) [inline HelloMixpanel/HelloMixpanelTests/MiscEdgeCaseTests.m]: does this really need a whole other file, why not put it part of the base tests?

Comment (pchien) [reply HelloMixpanel/HelloMixpanelTests/MiscEdgeCaseTests.m]: it didn't feel like a base test but didn't fit into any of our current files. was more like a one off

Comment (pchien) [reply HelloMixpanel/HelloMixpanelTests/MiscEdgeCaseTests.m]: actually now that i'm looking MixpanelBaseTests doesn't actually have any tests at all. not sure if putting tests in there would run in all the subclasses

Comment (yarneo) [reply HelloMixpanel/HelloMixpanelTests/MiscEdgeCaseTests.m]: Here should be fine: https://github.com/mixpanel/mixpanel-iphone/blob/master/HelloMixpanel/HelloMixpanelTests/HelloMixpanelTests.m

Comment (yarneo) [reply HelloMixpanel/HelloMixpanelTests/MiscEdgeCaseTests.m]: i think this doesn't deserve a separate file especially when it's called MiscEdgeCaseTests

Comment (pchien) [reply HelloMixpanel/HelloMixpanelTests/MiscEdgeCaseTests.m]: ah i'll move it there. didn't realize that was already a mix of different tests

**Pull request merged**